Readme.md

```diff
@@ -1,21 +1,146 @@
+    - Representing the identity of the DAO (ENS name, logo, description, other metadata)
+    - Holding and managing the treasury
+    - Executing arbitrary actions to:
+        - transfer assets
+        - call its own functions
+        - call functions in external contracts
+    - Providing general technical utilities (signature validation, callback handling)
+2. **The Permission Manager:** The permission manager is an integral part of any DAO and the center of our protocol architecture. It **manages permissions for the DAO** by specifying which addresses have permission to call distinct functions on contracts associated with your DAO.
+3. **Plugin Manager**: Any custom functionality can be added or removed through plugins, allowing you to **fully customize your DAO**. Some examples of plugins DAOs could install are:
+    - Governance (e.g., token voting, one-person one-vote)
+    - Asset management (e.g., ERC-20 or NFT minting, token streaming, DeFi)
+    - Membership (governing budget allowances, gating access, curating a member list)
+
+The following graphic shows how an exemplary DAO setup, where the
+
+![An examplary DAO setup](https://devs.aragon.org/assets/images/dao-plugin.drawio-7086d0911d25218097dae94665b1a7b1.svg)
+
+An examplary DAO setup showing interactions between the three core contract pieces triggered by different user groups: The `DAO` and `PermissionManager` contract in blue and red, respectively, as well as two `Plugin` contracts in green. Function calls are visualized as black arrows and require permission checks (red, dashed arrow). In this example, the permission manager determines whether the token voting plugin can execute actions on the DAO, a member can change its settings, or if an DeFi-related plugin is allowed to invest in a certain, external contract.
+
+### Framework Contracts
+
+In contrast, the *Framework Contracts* are in charge of creating and registering DAOs and plugins. Additionally, it’s what holds the plugin installer to install or uninstall plugins into DAOs.
+
+- __Factories and Registries__
+    - **The DAO Factory**: In charge of deploying instances of a new DAO based on the parameters given, including which plugins to install and additional metadata the DAO has (like a name, description, etc).
+    - **The DAO Registry**: In charge of registering DAOs into our protocol so we can easily access all DAO instances within our protocol. It is also in charge of giving DAOs subdomains for easier access.
+    - **The Plugin Repo Factory**: A `PluginRepo` is the repository of versions for a given plugin. The Plugin Repo Factory contract creates `PluginRepo` instances so that plugins can update their contract versioning without complexity in a semantic way similar to the App Store.
+    - **The Plugin Repo Registry**: In charge of registering the `PluginRepo` instances so that we easily access all protocol plugins.
```

Comment:
I would simply call it "plugin registry". It's true that it contains pluginRepo's but might confuse people

Reply:
Ok - calling these `Plugin Registry` and `Plugin Factory`, but still explaining the `PluginRepo` I think is important because otherwise the dev will go into the folders and not understand what they are/why they're there.